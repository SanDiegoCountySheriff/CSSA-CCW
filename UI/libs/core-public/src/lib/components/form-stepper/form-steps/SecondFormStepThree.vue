<template>
  <div>
    <ApplicationInfoSection />
    <v-form>
      <v-subheader class="sub-header font-weight-bold">
        {{ $t('Application Type') }}
      </v-subheader>
      <div class="ml-5">
        <RadioGroupInput
          :layout="'column'"
          :options="[
            { label: 'Standard', value: 'standard' },
            { label: 'Judicial', value: 'judicial', color: 'warning' },
            { label: 'Reserve', value: 'reserve', color: 'warning' },
          ]"
          @input="
            v => {
              completeApplicationStore.completeApplication.applicationType = v;
              state.valid = true;
            }
          "
        />
      </div>
      <v-alert
        dense
        outlined
        type="warning"
        v-if="
          completeApplicationStore.completeApplication.applicationType ===
          'judicial'
        "
      >
        <strong>
          {{ $t('Judicial-warning') }}
        </strong>
      </v-alert>
      <v-alert
        dense
        outlined
        type="warning"
        v-if="
          completeApplicationStore.completeApplication.applicationType ===
          'reserve'
        "
      >
        <strong>
          {{ $t('Judicial-reserve') }}
        </strong>
      </v-alert>
    </v-form>
    <v-divider />
    <FormButtonContainer
      :valid="state.valid"
      @submit="updateMutation.mutate"
      @save="saveMutation.mutate"
    />
  </div>
</template>

<script setup lang="ts">
import ApplicationInfoSection from '@shared-ui/components/info-sections/ApplicationInfoSection';
import FormButtonContainer from '@core-public/components/containers/FormButtonContainer.vue';
import RadioGroupInput from '@shared-ui/components/inputs/RadioGroupInput.vue';
import { reactive } from 'vue';
import { useCompleteApplicationStore } from '@core-public/stores/completeApplication';
import { useRouter } from 'vue-router/composables';
import { i18n } from '@shared-ui/plugins';
import { useMutation } from '@tanstack/vue-query';

interface ISecondFormStepThreeProps {
  handleNextSection: CallableFunction;
}

const props = defineProps<ISecondFormStepThreeProps>();
const completeApplicationStore = useCompleteApplicationStore();
const router = useRouter();

const state = reactive({
  valid: false,
});

const updateMutation = useMutation({
  mutationFn: () => {
    return completeApplicationStore.updateApplication('Step seven complete');
  },
  onSuccess: () => {
    props.handleNextSection();
  },
  onError: error => {
    alert(error);
  },
});

const saveMutation = useMutation({
  mutationFn: () => {
    return completeApplicationStore.updateApplication('Save and quit');
  },
  onSuccess: () => {
    router.push('/');
  },
  onError: () => {
    alert(i18n.t('Save unsuccessful, please try again'));
  },
});
</script>

<style lang="scss" scoped></style>
