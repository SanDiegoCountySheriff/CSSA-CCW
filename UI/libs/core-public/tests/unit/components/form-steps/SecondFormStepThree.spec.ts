import SecondFormStepThree from '@core-public/../../../../../shared/ui/src/lib/components/form-stepper/form-steps/SecondFormStepThree.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

const pinia = createTestingPinia();
const localVue = createLocalVue();
const tMock = {
  $t: text => text,
};

describe('SecondFormStepThree', () => {
  let vuetify;
  let wrapper;

  beforeEach(() => {
    vuetify = new Vuetify();
    wrapper = mount(SecondFormStepThree, {
      localVue,
      vuetify,
      pinia,
      mocks: tMock,
      propsData: {
        handleNextSection: () => null,
      },
      data() {
        return {
          valid: false,
        };
      },
    });
  });

  it('Should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
