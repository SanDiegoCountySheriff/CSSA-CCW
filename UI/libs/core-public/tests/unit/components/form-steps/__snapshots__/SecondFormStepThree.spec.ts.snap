// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondFormStepThree Should match the snapshot 1`] = `
<div>
  <div class="v-card v-sheet theme--light">
    <div class="v-card__title">
      Application-Type-Header
    </div>
    <p class="text-left ml-5">
      Application-Type-Subheader
    </p>
    <div class="v-card__text">
      <div class="text-line">
        <h3 class="text-label">
          STANDARD
        </h3>
        <p class="text-info">
          Application-Standard
        </p>
      </div>
      <div class="text-line">
        <h3 class="text-label">
          JUDICIAL-title
        </h3>
        <p class="text-info">
          Application-Type-Judicial
        </p>
      </div>
      <div class="text-line">
        <h3 class="text-label">
          RESERVE-title
        </h3>
        <p class="text-info">
          Application-Type-Reserve
        </p>
      </div>
    </div>
  </div>
  <form novalidate="novalidate" class="v-form">
    <div class="v-subheader sub-header font-weight-bold theme--light">
      Application Type
    </div>
    <div class="ml-5">
      <div class="v-input theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column">
        <div class="v-input__control">
          <div class="v-input__slot" style="height: auto;">
            <div role="radiogroup" aria-labelledby="input-6" class="v-input--radio-group__input">
              <div class="v-radio theme--light">
                <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-7" role="radio" type="radio" name="radio-6" value="standard">
                  <div class="v-input--selection-controls__ripple"></div>
                </div><label for="input-7" class="v-label theme--light" style="left: 0px; position: relative;">Standard</label>
              </div>
              <div class="v-radio theme--light">
                <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-9" role="radio" type="radio" name="radio-6" value="judicial">
                  <div class="v-input--selection-controls__ripple"></div>
                </div><label for="input-9" class="v-label theme--light" style="left: 0px; position: relative;">Judicial</label>
              </div>
              <div class="v-radio theme--light">
                <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-11" role="radio" type="radio" name="radio-6" value="reserve">
                  <div class="v-input--selection-controls__ripple"></div>
                </div><label for="input-11" class="v-label theme--light" style="left: 0px; position: relative;">Reserve</label>
              </div>
            </div>
          </div>
          <div class="v-messages theme--light">
            <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
          </div>
        </div>
      </div>
    </div>
    <!---->
    <!---->
  </form>
  <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
  <div class="form-btn-container">
    <div class="form-btn-inner">
      <div><button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small success "><span class="v-btn__content">
        Continue
      </span></button> <button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small accent mx-2"><span class="v-btn__content">
        Save and Exit
      </span></button></div>
      <div><button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small warning mx-2"><span class="v-btn__content">
        Go back
      </span></button> <button type="button" class="mt-3 v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--small error mr-2"><span class="v-btn__content">
        Cancel
      </span></button></div>
    </div>
  </div>
  <div class="v-snack v-snack--bottom">
    <transition-stub name="v-snack-transition">
      <div class="v-snack__wrapper v-sheet v-sheet--outlined theme--light error--text" style="display: none;">
        <div role="status" aria-live="polite" class="v-snack__content">
          Section update unsuccessful please try again.
        </div>
        <div class="v-snack__action "></div>
      </div>
    </transition-stub>
  </div>
</div>
`;
